src/main/java/com/minecolonies/coremod/colony/ColonyManager.java

```diff
@@ -178,10 +199,116 @@ public static Colony createColony(@NotNull final World w, final BlockPos pos, @N
+        Utils.checkDirectory(chunkDir);
+
+        for(int i = chunkX - maxRange; i <= chunkX + maxRange; i++)
+        {
+            for(int j = chunkZ - maxRange; j <= chunkZ + maxRange; j++)
+            {
+                if(i >= chunkX - range && j >= chunkZ - range && i <= chunkX + range && j <= chunkZ + range)
+                {
+                    @NotNull final File file = new File(chunkDir, String.format(FILENAME_CHUNK, i, j, dimension));
+                    @NotNull final ChunkLoadStorage newStorage = new ChunkLoadStorage(id, ChunkPos.asLong(i,j), add, dimension);
+                    if (file.exists())
+                    {
+                        @Nullable final NBTTagCompound chunkData = loadNBTFromPath(file);
+                        final ChunkLoadStorage storage = new ChunkLoadStorage(chunkData);
+                        storage.merge(newStorage);
+
+                        if(storage.isEmpty())
+                        {
+                            file.delete();
+                        }
+                        else
+                        {
+                            saveNBTToPath(file, newStorage.toNBT());
+                        }
+                    }
+                    else
+                    {
+                        saveNBTToPath(file, newStorage.toNBT());
+                    }
+                }
+            }
+        }
+        missingChunksToLoad+= maxRange * maxRange;
+    }
+
+    /**
+     * Add a chunk storage to a chunk.
+     * @param chunk the chunk to add it to.
+     * @param storage the said storage.
+     */
+    public static void addStorageToChunk(final Chunk chunk, final ChunkLoadStorage storage)
+    {
+        final IColonyTagCapability cap = chunk.getCapability(CLOSE_COLONY_CAP, null);
+        storage.applyToCap(cap);
+        Log.getLogger().warn("Loading Chunk: " + chunk.x + " " + chunk.z);
+        chunk.markDirty();
+        MineColonies.getNetwork().sendToAll(new UpdateChunkCapabilityMessage(cap, chunk.x, chunk.z));
+    }
+
+    /**
+     * Load the colony info for a certain chunk.
+     * @param chunk the chunk.
+     * @param world the worldg to.
+     */
+    public static void loadChunk(final Chunk chunk, final World world)
+    {
+        if(missingChunksToLoad > 0)
+        {
+            @NotNull final File chunkDir = new File(DimensionManager.getWorld(0).getSaveHandler().getWorldDirectory(), CHUNK_INFO_PATH);
+            @NotNull final File file = new File(chunkDir, String.format(FILENAME_CHUNK, chunk.x, chunk.z, world.provider.getDimension()));
+            if(file.exists())
+            {
+                @Nullable final NBTTagCompound chunkData = loadNBTFromPath(file);
+                final ChunkLoadStorage storage = new ChunkLoadStorage(chunkData);
+                addStorageToChunk(chunk, storage);
+                file.delete();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Do something with the "boolean" value returned by "delete". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS899)